### PR TITLE
feat!: keep Item created dates when resupplying TDE-1298

### PR DESCRIPTION
**Pull request by paulfouquet**
### Motivation

- Ensure that the STAC Item `assets.visual.created` and `properties.created` datetimes are not changed when resupplying.
- Ensure that the STAC Item `assets.visual.updated` and `properties.updated` datetimes all are controlled by the caller of the standardise+validate job workflow. This ensures that all the `updated` datetimes for the entire (re-)supply are the same.

### Modifications

- Add `--current-datetime` parameter to standardise+validate (RFC 3339 UTC datetime).
- Keep `created` dates the same on resupply.
- Only change asset `updated` datetime when its checksum changes.
- Cache GDAL version response. This can be simplified to be an environment variable, to avoid having to shell out to GDAL.

### Verification

Unit/integration/E2E tests.

### CODE REVIEW

**Review by l0b0 — DISMISSED** [review]

**Comment by l0b0** [comment]
Approving as pair programmer.

**Review by l0b0 — DISMISSED** [review]

File: scripts/stac/imagery/create_stac.py [hunk 1]
```diff
@@ -127,48 +126,75 @@ def create_item(
...
+            "processing:version": environ.get("GIT_VERSION", "GIT_VERSION not specified"),
+        }
+    )
+
+    id_ = get_file_name_from_path(asset_path)
+    file_content = fs.read(asset_path)
+    created_datetime = updated_datetime = current_datetime
+
+    if published_path:
+        # FIXME: make this try/catch nicer
```
Comment (schmidtnz) [inline scripts/stac/imagery/create_stac.py:166]: does this need to be addressed before merging?

File: scripts/stac/imagery/tests/generators.py [hunk 2]
```diff
@@ -32,3 +36,11 @@ def any_stac_processing() -> STACProcessing:
             "processing:version": "any processing version",
         }
     )
+
+
+def any_multihash_as_hex() -> str:
+    return multihash_as_hex(os.urandom(64))
+
+
+def any_gdal_version() -> str:
```
Comment (schmidtnz) [inline scripts/stac/imagery/tests/generators.py:45]: this doesn't appear to get used

File: scripts/stac/imagery/tests/generators.py [hunk 3]
```diff
@@ -1,7 +1,11 @@
+import os
```
Comment (schmidtnz) [inline scripts/stac/imagery/tests/generators.py:1]: more specific import?

**Review by schmidtnz — COMMENTED** [review]
Mostly nitpicking.
As a more general comment, it looks like harmonizing how we handle values and data for testing might potentially make our code more readable and maintainable.

File: scripts/stac/imagery/create_stac.py [hunk 4]
```diff
@@ -127,48 +126,76 @@ def create_item(
...
+    created_datetime = updated_datetime = current_datetime
+
+    if published_path:
+        try:
+            existing_item_content = read(path.join(published_path, f"{id_}.json"))
+            existing_item = json.loads(existing_item_content.decode("UTF-8"))
+            try:
+                created_datetime = existing_item["properties"]["created"]
+            except KeyError:
+                get_log().info(f"Existing Item {id_} does not have 'properties.created' attribute")
```
Comment (l0b0) [inline scripts/stac/imagery/create_stac.py:172]: In this case we should not set the created datetime at all - it needs to be backfilled.

**Comment by schmidtnz** [comment]
Closing this now that TDE-1298 has been merged through a new branch. @paulfouquet

**Pull request closed without merging**